pkg/gadgets/trace/tcpconnlat/tracer/gadget.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracer
+
+import (
+	gadgetregistry "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-registry"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/tcpconnlat/types"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/parser"
+)
+
+const (
+	ParamMin = "min"
+)
+
+type GadgetDesc struct{}
+
+func (g *GadgetDesc) Name() string {
+	return "tcpconnlat"
+}
+
+func (g *GadgetDesc) Category() string {
+	return gadgets.CategoryTrace
+}
+
+func (g *GadgetDesc) Type() gadgets.GadgetType {
+	return gadgets.TypeTrace
+}
+
+func (g *GadgetDesc) Description() string {
+	return "Trace TCP connects and show connection latency"
+}
+
+func (g *GadgetDesc) ParamDescs() params.ParamDescs {
+	return params.ParamDescs{
+		{
+			Key:          ParamMin,
+			Title:        "min",
+			DefaultValue: "0",
+			Description:  "Show only connect events with higher latency than min µs",
+			TypeHint:     params.TypeUint64,
```

Comment:
I am wondering about using `params.TypeString` and accepting a duration with the syntax described in:
https://pkg.go.dev/time#ParseDuration

That would be more friendly.

Reply:
Added this support to the params library.